### DISCUSSION TITLE
Userscripts IDEAS

### DISCUSSION THREAD

**Discussion by AceOfSettlers** [fedorovvl/tso_client#59] [Ideas]
The Scripts IDEAS Area

**Comment by Oviz** [comment]
Hi all, looking for script like: BATTLE RECORD
(with units refill)

**Reply by internety** [reply]
what you mean 'with units refill'? what does units refill have to do with battle record? :D

**Comment by Oviz** [comment]
Script: Tools -> Generals 
I think need button (load army) 
If i saved with special army, i want to load general with army :)

**Comment by hoanhtuan-hat** [comment]
I think we can save and load these steps:

- Army pre-configuration for an adventure, if not enough soldiers then pup up alert, if enough then send to adventure
Have a menu/list to do SINGLE step attack:
+ Attack G1(any name of attack). Send Gen X + Gen Y
+ Attack G3(any name of attack). Send Gen K + Gen Z.....
Players just need to record once and do step-by-step later

**Comment by X-Xadro** [comment]
How about an overview of all Treasure/Adventure Loot mails, i always wondered how much those 100+ explorers bring back, especially during events.

**Reply by internety** [reply]
I'm afraid that treasure messages doesn't contain any reference to specialist that gathered that trasure, except of skills used during searching... Does it?

**Comment by l23l** [comment]
Automatic addition of treasures to star/magazine from message window

**Comment by MrBadger** [comment]
Hi, and thanks to all for the great work here.

I wondered about the possibility of a script that lists all buildings and decorations etc on the island and allows you to click on them to select/locate them. The economy overview does this for production buildings but not for others.  I realise it would be a long list so perhaps group in drop down type categories if possible. 

Thanks.

**Comment by igorSof** [comment]
Хорошо бы какой-то учет всех ресурсов зданий специалистов вести, как простой вариант сохранять html файл с таблицей со всем хозяйством игрока. как сложный вариант организовать где-то на сайте мерялку богатств игроков, понятно дело добровольно самим игроком выгружать в открытый доступ свои сбережения.

**Comment by Anprivate** [comment]
А можно ли при отсылке геологов проверять - будет ли им что найти? Что я имею в виду:
Выбираю, например, 7 charming'ов для поиска titanium. После выбора (или по нажатию отдельной кнопки) внутри создаётся список всех exhausted и работающих titanium шахт, рассчитывается время до их истощения и сравнивается со временем поиска выбранного геолога. Ну и рассчитывается - сколько будет exhausted шахт к моменту окончания поиска и если их меньше, чем количество выбранных геологов, то помечать "лишних" геологов красным, например. Это существенно упростит "раннюю" отсылку геологов - то есть их отправку до того, как ресурс будет исчерпан с расчетом времени исчерпания.
Сейчас, например, у меня есть большая таблица в excel, в которую забито время цикла для каждой шахты и рассчитано - при каком оставшемся уровне ресурса можно отправлять геолога. Понятно, что это громоздко, неудобно и т.д. Вот бы это автоматизировать

**Reply by fedorovvl** [reply]
мне кажется вы можете перед отправкой смотреть в скрипт "Прелестные залежи" и оценивать сколько геологов нужно посылать.. там как раз есть инфа по времени истощения

**Reply by Anprivate** [reply]
Можно. Но.
1.  Там показывается время окончания, а не время до окончания. То есть надо предварительно рассчитать время возврата геолога и ориентироваться на него
2. Не учитываются уже посланные геологи.
То есть поэтапная отправка геологов становится тем еще развлечением (посмотри сколько геологов уже ищут этот ресурс, посмотри сколько осталось свободных геологов, посмотри время окончания поиска этими геологами, перейди на скрипт charming ddeposits, посмотри сколько депозитов кончится к этому времени, посчитай сколько можно отправить геологов с учетом уже отправленных, перейди в скрипт геологов, отправь нужное количество геологов).
Вот всего этого и хотелось бы избежать :)

**Comment by krajanos** [comment]
I think it would be nice to have another script for open tasks.
For example:
You have task to buy/sell 200 marble. With one click you can select friend who will receive offer (or two way offers).

**Comment by Anprivate** [comment]
А, вот еще идея. Можно ли при отсылке эксплореров сделать контроль крайнего времени прибытия? То есть в некое поле вводим, например, 23:00. И все эксплореры, возвращающиеся после этого времени, помечаются красным. Это было бы очень полезно во время event'ов, когда отсылаешь по несколько десятков эксплореров и вынужден всё время контролировать - вернутся ли они до того, как ляжешь спать, или надо выбрать посыл покороче или наоборот подольше, чтобы они уже к утру вернулись.
Понятно, что можно просто по времени смотреть, но когда их несколько десятков, то можно что-то пропустить.

**Reply by Urvdmih** [reply]
У меня еще есть идея, совместить основой модуль исследователей со скриптом Spec Duty. Чтобы те которые прохлаждаются наверху были, а остальные списком ниже, кто на задании. Чтобы сразу видно было, до какого времени отправить можно, чтобы плюс минус в одно время пришли.

**Comment by SirAdams** [comment]
I am looking for a script that would show me a list of buildings on an island and when I click on it, it would show me where it is located. 
Sometimes I wonder if I have a building and if I do where it is. It would make the game easier for me.

**Reply by krajanos** [reply]
> I am looking for a script that would show me a list of buildings on an island and when I click on it, it would show me where it is located. Sometimes I wonder if I have a building and if I do where it is. It would make the game easier for me.

I think there is something useful for you, have you seen userscript 'user_deposit.js'?

**Reply by SirAdams** [reply]
Yes, something like that only that this script shows resource-producing buildings and I would like to find, for example, "Toy Factory" or "Concert Hall" etc.

**Reply by skelgaard** [reply]
> Yes, something like that only that this script shows resource-producing buildings and I would like to find, for example, "Toy Factory" or "Concert Hall" etc.

you mean like F8 ? it just needs the links added :)

**Comment by panthibotronix** [comment]
> The Scripts IDEAS Area

Hi, it would be nice to have a script to integrate TSO chat with discord. It is possible?

**Reply by skelgaard** [reply]
thats a nice idea, and it proberly could be made, but only as long as a user is online on the game... it can't se if the user is offline

**Comment by Daveski-Memnon** [comment]
I had a thought about having a script that will remove resources from your star so that you do not do them 1 at a time, can anyone help with that?

**Comment by hercomandiren** [comment]
Идея блокировки/снятия зональных фильтров от НЕ друзей или дураков (сердечки, фейерверки, доска спиритов, прикормка для оленей, единороги, снег, ночь, солнечная погода)
Есть конечно фильтр по умолчанию, но он не спасает от фейерверков.
Я через консоль конечно пытался, собрал вот такие вот команды:
game.gi.mZoneBuffManager.removeBuff('555', 0) //Убирает Феерверки
game.gi.mZoneBuffManager.removeBuff('632', 0) //Убирает Сердечки
game.gi.mZoneBuffManager.removeBuff('41', 0) // Убирает фильтры солнце, ночь, снег
Но это все работает только локально и если перейти в другую зону, потом вернуться, или обновить зону все эти гадостные садизмы для глаз снова на месте.
Очень надеюсь что поддержат мою идею, потому что жутко раздражают эти фильтры, особенно в преддверии ивента валентина очень бы хотелось уже имень блокировку/автоснятие этих фильтров.

**Reply by Catavasiya** [reply]
Чего уж там, даешь:
1) "Черный список" зоналок, которые никто на остров наложить не может, как бы не хотел. 
2) "Белый список" зданий, которые запрещено бафать не-владельцу. 
3) "Супер-замок", защищающий остров от входа кого бы то ни было на указанное кол-во минут.

**Comment by Moonie-73** [comment]
I'm looking for a script that can generate a list of all my buffable buildings, so I can forward to my buff-partner for easy exchange Buff lists. Information is available in current script: "Buildinglist", but no export function there and you have to pick the individual building types.

**Reply by BasilMas** [reply]
You can create your own template with  Buildings -> Buffs and send it to your buff-partner. The list is assigned with player ID, and  anybody can use it for buffs  your  island.

**Reply by Catavasiya** [reply]
You can just send your Bufftemplate file to your friend (via e-mail, messengers etc) without any export and scripts.
![image](https://github.com/fedorovvl/tso_client/assets/128522616/ca60dee5-cf8e-4203-9a3e-bcbe6febc367)

**Reply by Moonie-73** [reply]
That requires me to buff my own island, so it will take forever, since I have to wait for current buffs to expire. Buff list is pretty simple, so info from from Buildinglist, with "buiGrid", "zoneId" and "zoneUser" would be more efficient. If you at the same time, manually could remove any unwanted buildings from the list and/or remove buildings currently being upgraded, could make it a great tool.

**Reply by BasilMas** [reply]
Sirris - the author of client - doesn't support the idea of 'full automation' in the game.  
You can join in our Discord server and create your request there - mb anybody can help you.

**Reply by Moonie-73** [reply]
Not interested in "full automation". Just a simple export function based on current "Building List" script, but without filter on buildingtypes. So a full list of buiGrid + Upgrading yes/no + info about buffable or not (if available).

Other stuff is simply "nice to have features". :)

**Reply by BasilMas** [reply]
Write to Fera - the author of the script "Building List" :)
But I use Buildings -> Production to create a list of buildings needed with further editing of template in Notepad++ manually

**Comment by Coffeeye** [comment]
Hey, Iam looking for script that will generate production in for example Provision House.
So I can make a schemat named "Storm Recovery" and by one click queue production in it based on generated schemat.
